deployments/helm-chart/templates/controller-daemonset.yaml

```diff
@@ -142,6 +142,7 @@ spec:
 {{- end }}
           - -enable-prometheus-metrics={{ .Values.prometheus.create }}
           - -prometheus-metrics-listen-port={{ .Values.prometheus.port }}
+          - -prometheus-tls-secret={{ .Values.prometheus.secret }}
```

Comment:
I haven't tested it (it needs to be checked with a deployed Prometheus) but the pod template is probably needed to be updated with `prometheus.io/scheme: https`  (both for Deployment and DS) if TLS is enabled for Prometheus endpoint

Reply:
You're correct, I didn't realise it but prometheus uses the annotations as a marker to scrape. Fixed.